core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -163,6 +163,8 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
 
       case StopExecutors =>
         logInfo("Asking each executor to shut down")
+        // a tricky way to stop YarnAllocator request containers
+        requestTotalExecutors(0, 0, Map.empty)
```

Comment:
Is it better to fix this in the `YarnAllocator` or `ApplicationMaster`? Change here will affect all the cluster manager, not only Yarn.